sencha-workspace/packages/slate-cbl/src/view/student/skill/OverviewWindowController.js

```diff
@@ -0,0 +1,86 @@
+/*jslint browser: true, undef: true *//*global Ext,Slate*/
+Ext.define('Slate.cbl.view.student.skill.OverviewWindowController', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.slate-cbl-student-skill-overviewwindow',
+    requires: [
+//        'Slate.cbl.API',
+//
+//        'Ext.MessageBox',
+//        'Ext.window.Toast'
+    'Slate.cbl.model.Competency',
+    'Slate.cbl.model.Skill'
+    ],
+
+    config: {
+        id: 'slate-cbl-student-skill-overviewwindow', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+                beforeshow: 'onBeforeWindowShow',
+                demorowclick: 'onDemoRowClick'
+            },
+            'combobox[reference=skillCombo]': {
+                change: 'onSkillChange'
+            }
+        }
+    },
+
+    // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+    applyId: function(id) {
+        return Ext.id(null, id);
+    },
+
+    onBeforeWindowShow: function(overviewWindow) {
+        var me = this,
+            competencyId = overviewWindow.getCompetency(),
+            student = overviewWindow.getStudent(),
+            skillsCombo = me.lookupReference('skillCombo'),
+            skillStore = skillsCombo.getStore(),
+            skillId = overviewWindow.getSkill();
+    
+        if (skillId && competencyId) {
+            
+            if(!skillStore.isLoaded()) {
+                Slate.cbl.API.getSkills(competencyId, function(response){
+                    skillStore.loadRawData(response.data);
+
+                    skillsCombo.setValue(skillStore.findRecord('ID', skillId));
+                    me.syncDemonstrationsTable();
+                });
+
+            } else {
+                me.syncDemonstrationsTable();
+            }          
+        }
+    },
+    
+    onDemoRowClick: function(overviewWindow, ev, targetEl) {
+        targetEl.next('.skill-grid-demo-detail-row').toggleCls('is-expanded');
+        overviewWindow.doLayout();
+    },
+
+    onSkillChange: function(skillCombo, skill) {
+        this.syncDemonstrationsTable();
```

Comment:
Also, it's  dangerous to plug events directly into member methods that weren't defined explicitly handle said event (i.e. prefixed with on and having a signature matching the events) as the member method may have parameters in its signature that don't align with the event's parameters, or optional parameters might be added to it later. For example imagine we later add a forceRefresh boolean to syncDemonstrationsTable, the skillCombo parameter would unexpectedly activate it.